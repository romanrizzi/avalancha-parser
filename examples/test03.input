
-- Invocaciones a funciones

check f() == f

check f(f) == g(H) -- un parámetro

-- varios parámetros
check a(b) == f(f)

-- Constructores
check zero() == zero()  -- sin parámetros (con paréntesis)

-- con un parámetro
check suc(zero) == suc(  zero  )

-- con varios parámetros
check a(b) == a(B)
check a(b,C) == a(B,c)
check a(b,C,D) == a(B,c,d)
check a(B,C,d,e) == a(B,c,d,e)

-- Recursivos
check a(b(c(d(e(f))))) == a(b(c(d(e(F)))))

check a(b(c),d(E,f(g))) == a(b(c),d(E,f(g)))

check sUC (
  sUC(zERO
)) == sUC(sUC(     zERO ) )

check cONS (a,cONS(b,cONS(
  c,conS(d,
  nIL)))) == cONS(a,cONS(b,cONS(c,cONS(d,      nIL))))

