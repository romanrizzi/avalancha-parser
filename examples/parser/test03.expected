[
   "program",
   [],
   [
      [
         "check",
         [
            "equal",
            [
               "app",
               "f",
               []
            ],
            [
               "var",
               "f"
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "app",
               "f",
               [
                  [
                     "var",
                     "f"
                  ]
               ]
            ],
            [
               "app",
               "g",
               [
                  [
                     "cons",
                     "H",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "app",
               "a",
               [
                  [
                     "var",
                     "b"
                  ]
               ]
            ],
            [
               "app",
               "f",
               [
                  [
                     "var",
                     "f"
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "app",
               "zero",
               []
            ],
            [
               "app",
               "zero",
               []
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "app",
               "suc",
               [
                  [
                     "var",
                     "zero"
                  ]
               ]
            ],
            [
               "app",
               "suc",
               [
                  [
                     "var",
                     "zero"
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "app",
               "a",
               [
                  [
                     "var",
                     "b"
                  ]
               ]
            ],
            [
               "app",
               "a",
               [
                  [
                     "cons",
                     "B",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "app",
               "a",
               [
                  [
                     "var",
                     "b"
                  ],
                  [
                     "cons",
                     "C",
                     []
                  ]
               ]
            ],
            [
               "app",
               "a",
               [
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "var",
                     "c"
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "app",
               "a",
               [
                  [
                     "var",
                     "b"
                  ],
                  [
                     "cons",
                     "C",
                     []
                  ],
                  [
                     "cons",
                     "D",
                     []
                  ]
               ]
            ],
            [
               "app",
               "a",
               [
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "var",
                     "c"
                  ],
                  [
                     "var",
                     "d"
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "app",
               "a",
               [
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "cons",
                     "C",
                     []
                  ],
                  [
                     "var",
                     "d"
                  ],
                  [
                     "var",
                     "e"
                  ]
               ]
            ],
            [
               "app",
               "a",
               [
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "var",
                     "c"
                  ],
                  [
                     "var",
                     "d"
                  ],
                  [
                     "var",
                     "e"
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "app",
               "a",
               [
                  [
                     "app",
                     "b",
                     [
                        [
                           "app",
                           "c",
                           [
                              [
                                 "app",
                                 "d",
                                 [
                                    [
                                       "app",
                                       "e",
                                       [
                                          [
                                             "var",
                                             "f"
                                          ]
                                       ]
                                    ]
                                 ]
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ],
            [
               "app",
               "a",
               [
                  [
                     "app",
                     "b",
                     [
                        [
                           "app",
                           "c",
                           [
                              [
                                 "app",
                                 "d",
                                 [
                                    [
                                       "app",
                                       "e",
                                       [
                                          [
                                             "cons",
                                             "F",
                                             []
                                          ]
                                       ]
                                    ]
                                 ]
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "app",
               "a",
               [
                  [
                     "app",
                     "b",
                     [
                        [
                           "var",
                           "c"
                        ]
                     ]
                  ],
                  [
                     "app",
                     "d",
                     [
                        [
                           "cons",
                           "E",
                           []
                        ],
                        [
                           "app",
                           "f",
                           [
                              [
                                 "var",
                                 "g"
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ],
            [
               "app",
               "a",
               [
                  [
                     "app",
                     "b",
                     [
                        [
                           "var",
                           "c"
                        ]
                     ]
                  ],
                  [
                     "app",
                     "d",
                     [
                        [
                           "cons",
                           "E",
                           []
                        ],
                        [
                           "app",
                           "f",
                           [
                              [
                                 "var",
                                 "g"
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "app",
               "sUC",
               [
                  [
                     "app",
                     "sUC",
                     [
                        [
                           "var",
                           "zERO"
                        ]
                     ]
                  ]
               ]
            ],
            [
               "app",
               "sUC",
               [
                  [
                     "app",
                     "sUC",
                     [
                        [
                           "var",
                           "zERO"
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "equal",
            [
               "app",
               "cONS",
               [
                  [
                     "var",
                     "a"
                  ],
                  [
                     "app",
                     "cONS",
                     [
                        [
                           "var",
                           "b"
                        ],
                        [
                           "app",
                           "cONS",
                           [
                              [
                                 "var",
                                 "c"
                              ],
                              [
                                 "app",
                                 "conS",
                                 [
                                    [
                                       "var",
                                       "d"
                                    ],
                                    [
                                       "var",
                                       "nIL"
                                    ]
                                 ]
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ],
            [
               "app",
               "cONS",
               [
                  [
                     "var",
                     "a"
                  ],
                  [
                     "app",
                     "cONS",
                     [
                        [
                           "var",
                           "b"
                        ],
                        [
                           "app",
                           "cONS",
                           [
                              [
                                 "var",
                                 "c"
                              ],
                              [
                                 "app",
                                 "cONS",
                                 [
                                    [
                                       "var",
                                       "d"
                                    ],
                                    [
                                       "var",
                                       "nIL"
                                    ]
                                 ]
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ]
   ]
]
